tests/unittest/test_helpers.py

```diff
@@ -166,6 +173,981 @@ def test_merge_configurations(self):
+        )
+
+        assert merged_result.get("username") == username2
+        assert merged_result.get("namespace") == namespace2
+        assert merged_result.get("kubeconfig") == kubeconfig
+        assert merged_result.get("context") == context
+        assert merged_result.get(conf) == value2
+
+    @patch("helpers.utils.yaml.safe_load")
+    @patch("builtins.open")
+    @patch("helpers.utils.pwd.getpwuid")
+    @patch("helpers.utils.os.getuid")
+    def test_get_defaults_from_kubeconfig(
+        self, mock_os_get_uid, mock_pwd_getpwuid, mock_open, mock_yaml_safe_load
+    ):
+        # mock logic
+        test_id = str(uuid.uuid4())
+        username = str(uuid.uuid4())
+        kubeconfig = str(uuid.uuid4())
+        context = str(uuid.uuid4())
+        token = str(uuid.uuid4())
+
+        mock_os_get_uid.return_value = 100
+        mock_pwd_getpwuid.return_value = [
+            str(uuid.uuid4()),
+            str(uuid.uuid4()),
+            str(uuid.uuid4()),
+            str(uuid.uuid4()),
+            str(uuid.uuid4()),
+            str(uuid.uuid4()),
+        ]
+
+        mock_yaml_safe_load.return_value = {
+            "apiVersion": "v1",
+            "clusters": [
+                {
+                    "cluster": {
+                        "certificate-authority-data": f"{test_id}",
+                        "server": f"https://0.0.0.0:{test_id}",
+                    },
+                    "name": f"{context}-cluster",
+                }
+            ],
+            "contexts": [
+                {
+                    "context": {"cluster": f"{context}-cluster", "user": f"{username}"},
+                    "name": f"{context}",
+                }
+            ],
+            "current-context": f"{context}",
+            "kind": "Config",
+            "preferences": {},
+            "users": [{"name": f"{username}", "user": {"token": f"{token}"}}],
+        }
+
+        # test logic
+        env_snap = os.environ.get("SNAP")
+        os.environ["SNAP"] = test_id
+
+        with patch("builtins.open", mock_open(read_data="test")):
+            result = helpers.utils.get_defaults_from_kubeconfig(kubeconfig, context)
+
+        if env_snap:
+            os.environ["SNAP"] = env_snap
+
+        assert result["context"] == context
+        assert result["namespace"] == "default"
+        assert result["cert"] == test_id
+        assert result["config"] == kubeconfig
+        assert result["user"] == "spark"
+
+    @patch("builtins.input")
+    def test_select_context_id(self, mock_input):
+        # mock logic
+        test_id = str(uuid.uuid4())
+        username1 = str(uuid.uuid4())
+        context1 = str(uuid.uuid4())
+        token1 = str(uuid.uuid4())
+        username2 = str(uuid.uuid4())
+        context2 = str(uuid.uuid4())
+        token2 = str(uuid.uuid4())
+        username3 = str(uuid.uuid4())
+        context3 = str(uuid.uuid4())
+        token3 = str(uuid.uuid4())
+
+        kubeconfig_yaml = {
```

Comment:
If may be best to store a YAML file in the tests folder and try to read from there. 

Also in this case, this does not need to be addressed now, but just keep in mind for the next tests restructuring